utils/communication_report.py

```diff
@@ -0,0 +1,133 @@
+'''
```

Comment:
since this is run as a standalone script, would probably make sense to add the shebang
```suggestion
#!/usr/bin/env python3
'''
```